src/Compilers/CSharp/Test/Semantic/Semantics/NativeIntegerTests.cs

```diff
@@ -6255,163 +6255,235 @@ .maxstack  2
-            constantFolding("nuint", "const nuint A = 1; const int B = 32;", "A >> B", "1");
-
-            constantFolding("nint", "const nint A = -2147483648; const nint B = 0;", "A & B", "0");
-            constantFolding("nint", "const nint A = -2147483648; const nint B = -1;", "A & B", "-2147483648");
-            constantFolding("nint", "const nint A = -2147483648; const nint B = 2147483647;", "A & B", "0");
-            constantFolding("nuint", "const nuint A = 0; const nuint B = 4294967295;", "A & B", "0");
-            constantFolding("nuint", "const nuint A = 2147483647; const nuint B = 4294967295;", "A & B", "2147483647");
-            constantFolding("nuint", "const nuint A = 2147483647; const nuint B = 2147483648;", "A & B", "0");
-
-            constantFolding("nint", "const nint A = -2147483648; const nint B = 0;", "A | B", "-2147483648");
-            constantFolding("nint", "const nint A = -2147483648; const nint B = -1;", "A | B", "-1");
-            constantFolding("nint", "const nint A = 2147483647; const nint B = 2147483647;", "A | B", "2147483647");
-            constantFolding("nuint", "const nuint A = 0; const nuint B = 4294967295;", "A | B", "4294967295");
-            constantFolding("nuint", "const nuint A = 2147483647; const nuint B = 2147483647;", "A | B", "2147483647");
-            constantFolding("nuint", "const nuint A = 2147483647; const nuint B = 2147483648;", "A | B", "4294967295");
-
-            constantFolding("nint", "const nint A = -2147483648; const nint B = 0;", "A ^ B", "-2147483648");
-            constantFolding("nint", "const nint A = -2147483648; const nint B = -1;", "A ^ B", "2147483647");
-            constantFolding("nint", "const nint A = 2147483647; const nint B = 2147483647;", "A ^ B", "0");
-            constantFolding("nuint", "const nuint A = 0; const nuint B = 4294967295;", "A ^ B", "4294967295");
-            constantFolding("nuint", "const nuint A = 2147483647; const nuint B = 2147483647;", "A ^ B", "0");
-            constantFolding("nuint", "const nuint A = 2147483647; const nuint B = 2147483648;", "A ^ B", "4294967295");
-
-            void constantFolding(string opType, string declarations, string expr, string expectedResult, DiagnosticDescription diagnostic = null)
+            const string intMinValue = "-2147483648";
+            const string intMaxValue = "2147483647";
+            const string uintMaxValue = "4294967295";
+            const string ulongMaxValue = "18446744073709551615";
+
+            unaryOperator("nint", "+", intMinValue, intMinValue);
+            unaryOperator("nint", "+", intMaxValue, intMaxValue);
+            unaryOperator("nuint", "+", "0", "0");
+            unaryOperator("nuint", "+", uintMaxValue, uintMaxValue);
+
+            unaryOperator("nint", "-", "-1", "1");
+            unaryOperatorCheckedOverflow("nint", "-", intMinValue, IntPtr.Size == 4 ? "-2147483648" : "2147483648");
```

Comment:
Still a bunch of magic constants in the tests. In this line I see `intMinValue` and what I assume is `intMaxValue+1`?